Source/Core/Gen5/Generators/HiddenGrottoGenerator.cpp

```diff
@@ -45,7 +45,7 @@ std::vector<HiddenGrottoState> HiddenGrottoGenerator::generate(u64 seed) const
     for (u32 cnt = 0; cnt <= maxAdvances; cnt++, rng.next())
     {
         BWRNG go(rng.getSeed());
-        u32 seed = go.nextUInt(0x1FFF);
+        u32 seed = ((go.getSeed() >> 32) * 0x1FFF) >> 32;
```

Comment:
This is correct in-game against chatot pitches?

Reply:
yes it is, i checked in game directly